### PR TITLE
v1.2.0 Added bower, composer and log libs

### PR DESCRIPTION
**Pull request by CZYK**
This should solve:
- https://github.com/studyportals/product-deploy/issues/9
- https://github.com/studyportals/product-deploy/issues/10

Please have a look at https://github.com/studyportals/product-deploy/blob/CZYK-Bower/README.md to see the proposed interface.

Note that the tests for bower and composer take quite some time (seconds). This will get annoying when we continue developing on this package and the amount of tests will grow. It would make sense to split of this functionality in it's own package. In the end there is also no need to re-test functionality which has not been touched every time. For now I will leave it in `product-deploy`.

### CODE REVIEW

File: lib/composer.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	let defaults = {
+		'cwd': process.cwd()
+	};
+
+	opts = Object.assign(defaults, opts);
+
+	return new Promise((resolve, reject) => {
+
+		fs.stat(path.resolve(opts.cwd, 'composer.json'), (err) =>{
```
Comment (iliqportals) [inline lib/composer.js]: same comment like bower

File: test/log.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"use strict";
+
+const log = require('./../lib/log');
+
+it('It should show all log messages', () =>{
+
+	log.setVerbosity(4);
```
Comment (iliqportals) [inline test/log.test.js]: this does not test anything.
Best to test the STDOUT if it contains the strings

File: test/use-strict.test.js [hunk 3]
```diff
@@ -10,7 +10,9 @@ describe('Structurally', function(){
 	it('all js files should "use strict"', () =>{
 
 		const nonStrictFiles = [];
-		const allFiles = globby.sync(["**/*.js", '!node_modules/**']);
+		const allFiles = globby.sync([
+			"index.js",
```
Comment (iliqportals) [inline test/use-strict.test.js]: everything under tests should be 'use strict' aswell

File: lib/log.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+const DEFAULT_VERBOSITY = 2;
+
+let verbosity = DEFAULT_VERBOSITY;
+
+/**
+ * Set the verbosity level
+ * - 0: No output
+ * - 1: Error output
+ * - 2: Warning output
```
Comment (iliqportals) [inline lib/log.js]: Would be a good practice to define these as Constants
...log.setVerbosity(log.Verbosity.None)
...log.setVerbisity(log,Verbosity.Warning)

File: lib/bower.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+const fs = require('fs');
+const bower = require('gulp-bower');
+const path = require('path');
+const log = require('./log');
+
+/**
+ * Install bower dependencies.
+ *
+ * @param {Object} [opts]
+ * @param {string} [opts.cwd=process.cwd()] current working directory
```
Comment (CZYK) [inline lib/bower.js]: > If none specified, uses process.cwd()

This is not needed as `[opts.cwd=process.cwd()]` already specifies that it is optional and what the default value is.

File: lib/bower.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+const fs = require('fs');
+const bower = require('gulp-bower');
+const path = require('path');
+const log = require('./log');
+
+/**
+ * Install bower dependencies.
+ *
```
Comment (CZYK) [inline lib/bower.js]: I rewrote it slightly
>  The installation will complete, without applying any changes, when no `bower.json` file is found.

File: lib/log.js (hunk 4 above)
Comment (CZYK) [inline lib/log.js]: Agreed. I will use UPPERCASE for the consts.

File: lib/log.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+/**
+ * Debug message (gray)
+ * @param {string} message
+ * @alias module:log
+ * @return void
+ */
+const debug = function(message){
+	if(verbosity > 3){
```
Comment (iliqportals) [inline lib/log.js]: Good to also use the constants
If Verbosity > VERBOSITY.INFO

File: lib/log.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+/**
+ * Info message (white)
+ * @param {string} message
+ * @alias module:log
+ * @return void
+ */
+const info = function(message){
+	if(verbosity > 2){
```
Comment (iliqportals) [inline lib/log.js]: using Verbosity constant

File: lib/log.js [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+/**
+ * Warning message (yellow)
+ * @param {string} message
+ * @alias module:log
+ * @return void
+ */
+const warning = function(message){
+	if(verbosity > 1){
```
Comment (iliqportals) [inline lib/log.js]: using Verbosity constant

File: lib/log.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+/**
+ * Error message (red)
+ * @param {string} message
+ * @alias module:log
+ * @return void
+ */
+const error = function(message){
+	if(verbosity > 0){
```
Comment (iliqportals) [inline lib/log.js]: using Verbosity constant

**Pull request merged**